File: 263.ugly-number.js
/*
 * @lc app=leetcode id=263 lang=javascript
 *
 * [263] Ugly Number
 */

// @lc code=start
/**
 * @param {number} num
 * @return {boolean}
 */
var isUgly = function (num) {
  if (num < 1) {
    return false;
  }
  while (num % 2 === 0) num /= 2;
  while (num % 3 === 0) num /= 3;
  while (num % 5 === 0) num /= 5;
  return num === 1;
};
// @lc code=end

File: 263.ugly-number.go
/*
 * @lc app=leetcode id=263 lang=golang
 *
 * [263] Ugly Number
 */

package leetcode

// @lc code=start
func isUgly(n int) bool {
	if n < 1 {
		return false
	}
	pf := [3]int{2, 3, 5}
	for i := 0; i < 3; i++ {
		for n%pf[i] == 0 {
			n /= pf[i]
		}
	}
	return n == 1
}

// @lc code=end
